Viktoriia_Kovalenko_JavaRush_Cryptoanalyser

cladyland opened the pull request:
My first Java project is ready to checking =)

cladyland committed:
Added class Alphabet

cladyland committed:
Added Artifacts, FileChecking, FileProcessing and FileProcessingException classes

cladyland committed:
Unsorted Alphabet changed to sorted

cladyland committed:
Added CaesarsCipher class

cladyland committed:
Added most_common_words.txt file

cladyland committed:
added inputIndex, outputIndex variables;
cipherKey variable changed from constant to normal;
changed caesarEncryptDecrypt method;
added findOutputIndex, commonWords and bruteForce methods

cladyland committed:
Rename commonWords method to mostCommonWords;
added symbolStatistics and statisticalAnalysis methods

cladyland committed:
Changed argument to create a new file in saveArtifact method;
added Dialog and Main classes

cladyland committed:
All classes and file 'most_common_words.txt' are moved to the appropriate folders

cladyland committed:
Added new variable to bruteForce method

cladyland committed:
Fixed dialogWithUser method

cladyland committed:
Added a test text file and a test file to save the result

cladyland committed:
Fixed error reading file in symbolStatistics method;
moved file most_common_words.txt

cladyland committed:
The result of the program was saved

cladyland committed:
Jar file created

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+package ua.com.javarush.vika_kovalenko.cryptoanalyser.files;
+
+import ua.com.javarush.vika_kovalenko.cryptoanalyser.model.Alphabet;
```
лучше избавится от структуры в виде ivan_ivanov - сделай лучше ivanov.ivan

cladyland replied:
ок, спасибо) исправляю..)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
```
лучше так не делать. потому что импортируется много чего лишнего

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+    }
+
+    public static void statisticalAnalysis() {
+        Optional<Map.Entry<Character, Integer>> maxEntry = symbolStatistics()
+                .entrySet()
+                .stream()
+                .max(Map.Entry.comparingByValue());
```
воу! класс)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+                    outputIndex -= ALPHABET_LENGTH;
+                }
+            }
+            case "encryptNegative", "decryptPositive" -> {
```
видел что это используется выше. тогда выноси в константу; какой шанс что ты написала это с первого раза и в след раз не ошибешься)

akupreychik replied:
я про encryptPositive  etc

cladyland replied:
ок, так и сделаю) спасибо)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+
+    private static HashSet<String> mostCommonWords() {
+        var hundredCommonWords = new HashSet<String>();
+        try (var reader = new BufferedReader(new FileReader("most_common_words.txt"))) {
```
тоже в константу most_common_words

cladyland replied:
я вызываю этот метод только в методе bruteForce() и думала, что константа будет лишней. Или так надо на случай, если еще придется где-то использовать этот Set?

cladyland replied:
или это про имя файла?)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+    private static HashSet<String> mostCommonWords() {
+        var hundredCommonWords = new HashSet<String>();
+        try (var reader = new BufferedReader(new FileReader("most_common_words.txt"))) {
+            for (int i = 0; i < 100; i++) {
```
а почему именно 100? тоже в константу

cladyland replied:
подумала, что из названия Set 'hundredCommonWords' должно быть понятно, почему именно 100))

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/model/Dialog.java:
```diff
@@ -0,0 +1,111 @@
+                                               "longer run.");
+        }
+        return filePath;
+    }
```
это не очень к диалогу относится)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/model/Dialog.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    private static void programMode(int modeNumber) {
+        switch (modeNumber) {
+            case 1 -> CaesarsCipher.encryption();
+            case 2 -> CaesarsCipher.decryption();
+            case 3 -> CaesarsCipher.bruteForce();
+            case 4 -> CaesarsCipher.statisticalAnalysis();
+        }
+    }
```
а если придет 5? добавь default для таких случаев)

cladyland replied:
у меня на такой случай есть предварительная проверка:
` if (userMode < 1 || userMode > 4) {
                    System.err.println(WRONG_MODE);
                }`
поэтому до default программа не сможет добраться. По крайней мере, не должна)

akupreychik commented on src/ua/com/javarush/vika_kovalenko/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,182 @@
+
+    public static void bruteForce() {
+        char[] symbols;
+        int numberOfLinesRequiredForVerification = 9;
```
можно и в константу. но это уже на твое рассмотрение. если ты используешь это только тут

cladyland replied:
да, это используется только в одном методе

akupreychik reviewed (commented):
Классная работа! Умничка!

cladyland committed:
Rename package 'vika_kovalenko' to 'kovalenko.vika' and moved all classes;
added constant variables to CaesarsCipher class

cladyland committed:
Replaced wildcard import with single class import;
added new variables;
class FileProcessing is no longer dependent on the Artifacts class;
moved preliminaryPathCheck method

cladyland committed:
New jar file created

AndryLat commented on src/ua/com/javarush/kovalenko/vika/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,193 @@
+                continue;
+            }
+            for (char symbol : symbols) {
+                if (symbol == '[' || symbol == ']') {
```
почему именно эти символы? если вынести в константу и дать норм имя, думаю, будет лучше и понятнее.

AndryLat commented on src/ua/com/javarush/kovalenko/vika/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,193 @@
+
+    private static void caesarEncryptDecrypt(String argument) {
+        for (String line : INPUT_LINES) {
+            char[] temp = line.toCharArray();
```
имя переменной ни о чем не говорит. да и можно отдельную переменную не создавать было.

cladyland replied:
и правда можно обойтись без этой переменной)) спасибо)

AndryLat commented on src/ua/com/javarush/kovalenko/vika/cryptoanalyser/files/CaesarsCipher.java:
```diff
@@ -0,0 +1,193 @@
+    private static HashSet<String> mostCommonWords() {
+        var hundredCommonWords = new HashSet<String>();
+        try (var reader = new BufferedReader(new FileReader(MOST_COMMON_WORDS))) {
+            int hundredWords = 100;
```
странно выглядит. Почему именно 100?

AndryLat replied:
topAmount, например.

cladyland replied:
thank you 👍

AndryLat commented on src/ua/com/javarush/kovalenko/vika/cryptoanalyser/model/Dialog.java:
```diff
@@ -0,0 +1,96 @@
+        var fileChecking = new FileChecking(filePath);
+        if (!fileChecking.isValidFilePath()) {
+            System.err.println(WRONG_FILE_PATH);
+            System.exit(1);
```
не надо в любой понятной ситуации вырубать приложение. корректный путь: выйти из main() метода. все остальное - это грязные хаки.

AndryLat reviewed (commented):
В целом хороший проект, статический контекст режет глаз, но это только во втором модуле будем исправлять, так что все круто)

cladyland committed:
Changed variables;
replaced the exit from the program in isFile method with an exception

cladyland committed:
New jar file created
